.ci/Jenkinsfile

```diff
@@ -0,0 +1,106 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+@Library('apm@current') _
+
+pipeline {
+  agent none
+  environment {
+    REPO = 'apm'
+    BASE_DIR = "src/github.com/elastic/${env.REPO}"
+    NOTIFY_TO = credentials('notify-to')
+    JOB_GCS_BUCKET = credentials('gcs-bucket')
+    PIPELINE_LOG_LEVEL='INFO'
+  }
+  options {
+    timeout(time: 3, unit: 'HOURS')
+    buildDiscarder(logRotator(numToKeepStr: '5', artifactNumToKeepStr: '5'))
+    timestamps()
+    ansiColor('xterm')
+    disableResume()
+    durabilityHint('PERFORMANCE_OPTIMIZED')
+  }
+  stages {
+    stage('Initializing'){
+      agent { label 'linux && immutable' }
+      options { skipDefaultCheckout() }
+      environment {
+        HOME = "${env.WORKSPACE}"
+        PATH = "${env.PATH}:${env.WORKSPACE}/bin:${env.WORKSPACE}/${env.BASE_DIR}/.ci/scripts"
+      }
+      stages {
+        stage('Checkout'){
+          agent { label 'immutable && docker' }
+          options { skipDefaultCheckout() }
+          steps {
+            deleteDir()
+            dir("${BASE_DIR}"){
+              checkout scm
+              stash allowEmpty: true, name: 'source', useDefaultExcludes: false
+            }
+          }
+        }
+        stage('Send Pull Request with APM specs to the APM Agents'){
+          agent { label 'immutable && docker' }
+          options {
+            skipDefaultCheckout()
+            warnError('Pull Requests to APM agents failed')
+          }
+          steps {
+            unstash 'source'
+            dir("${BASE_DIR}"){
+              script {
+                def agents = readYaml(file: '.ci/.jenkins-agents.yml')
```

Comment:
Let's use the matrix step if possible, for instance:
- https://github.com/elastic/ecs-logging-python/blob/63c4d889a68aae855b6dc5ccd6a56b6947a11254/.ci/Jenkinsfile#L51-L78

Reply:
Is it possible to define multiple values for the same axis? Otherwise I need to pass agent name and specific path as a string to be parsed (unfortunately each agent uses different paths to store the feature files).

As an example:
```groovy
       axes {
          axis {
            name 'AGENT'
            values 'python tests/bdd/features', 'ruby features', 'java apm-agent-core/src/test/resources/specs/api_key.feature'
          }
        }
```

Reply:
I missed that, it's not possible to specify a complex data structure but a list of values :(, so let's keep it as it is then :)